app/services/membership_modifier.rb

```diff
@@ -0,0 +1,38 @@
+class MembershipModifier
+  class ExistingMembership < StandardError; end
+  class NoExistingMembership < StandardError; end
+
+  def self.create_membership(requesting_user:, joinable:, target_user:, admin:)
+    return unless joinable.admin? requesting_user
```

Comment:
Loving the custom errors! Should we raise one here for `NotAdmin` or permitted or something? Just so it doesn't fail silently in case we forget and are stuck down the line wondering why something's not working?